lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -928,12 +971,16 @@ internal class EvaluatingCompiler(
                                 selectProjectionThunk(projectEnv, joinedValues)
                             }
 
-                            val quantifiedRows = when(setQuantifier) {
+                            var quantifiedRows = when(setQuantifier) {
                                 // wrap the ExprValue to use ExprValue.equals as the equality
                                 SetQuantifier.DISTINCT -> projectedRows.filter(createUniqueExprValueFilter())
                                 SetQuantifier.ALL -> projectedRows
                             }
 
+                            if (limit != null) {
+                                quantifiedRows = quantifiedRows.take(compileLimit(limit, env))
+                            }
```

Comment:
Can it be done using `let` to avoid using `var`?

```suggestion
                            val quantifiedRows = when(setQuantifier) {
                                // wrap the ExprValue to use ExprValue.equals as the equality
                                SetQuantifier.DISTINCT -> projectedRows.filter(createUniqueExprValueFilter())
                                SetQuantifier.ALL -> projectedRows
                            }.let { rows ->
                                when (limit) {
                                    null -> rows
                                    else -> rows.take(compileLimit(limit, env))
                                }
                            }
```

Reply:
+1

Reply:
Thanks, updated w/ the suggestion.